### DISCUSSION TITLE
isync/mbsync cannot connect to outlook.office365.com

### DISCUSSION THREAD

**Discussion by AaronBWagner** [microsoft/WSL#5898] [General] [answered]
Under Ubuntu 20.04 LTS on WSL2, mbsync (provided by the isync package) cannot connect to outlook.office365.com. Specifically, running

` sudo apt install isync; mbsync -V work`

with the skeleton ~/.mbsyncrc file:

>IMAPStore work-remote                                                                                                    
>  Host outlook.office365.com                                                                                            
>  Port 993                                                                                                               
>  SSLType IMAPS                                                                                                           
> SSLVersions TLSv1
>MaildirStore work-local 
> Path ~/Maildir/              
> Inbox ~/Maildir/INBOX                                                                                                   
> SubFolders Verbatim
> Channel work                                                                                                               
> Master :work-remote:                                                                                                   
>  Slave :work-local:                                                                                                     
>  Patterns "INBOX"   

in place yields the output:

> Reading package lists... Done  
> Building dependency tree                                                             
> Reading state information... Done                                                        
> Suggested packages:                                                                      
>   mutt                                                                                  
> ...                              
>Unpacking isync (1.3.0-2) ...                                                            
>Setting up isync (1.3.0-2) ...                                                           
>Processing triggers for man-db (2.9.1-1) ...                                             
>Reading configuration file /home/abw35/.mbsyncrc                                        
>C: 0/1  B: 0/0  M: +0/0 *0/0 #0/0  S: +0/0 *0/0 #0/0                                     
>Channel work                                                                             
>Opening master store work-remote...                                                      
>Resolving outlook.office365.com... ok                                                   
>Connecting to outlook.office365.com (52.96.79.98:993)...                                
>Opening slave store work-local...                                                        
>Socket error: secure connect to outlook.office365.com (52.96.79.98:993): Success        
>C: 1/1  B: 0/0  M: +0/0 *0/0 #0/0  S: +0/0 *0/0 #0/0     '`

Compiling isync-1.3.3 from source on Ubuntu 20.04 LTS/WSL2 yields the same behavior.
The expected output from mbsync is:

>Reading configuration file /Users/abw35/.mbsyncrc
 >C: 0/1  B: 0/0  M: +0/0 *0/0 #0/0  S: +0/0 *0/0 #0/0
>Channel work                                                                                                          
> Opening master store work-remote...           
 >Resolving outlook.office365.com... ok 
>Connecting to outlook.office365.com (52.96.15.178:993)...           
 >Opening slave store work-local... 
>Connection is now encrypted 
>Logging in...                                                                                                          
> Authenticating with SASL mechanism PLAIN...                   
 >Skipping account work-remote, no user                                                                                  
> C: 1/1  B: 0/0  M: +0/0 *0/0 #0/0  S: +0/0 *0/0 #0/0  

The above .mbsyncrc file contains no username or password lines because these are not necessary to expose a discrepancy between the expected and actual behavior. If I replace .mbsyncrc with my complete config file, I get the same output on Ubuntu/WSL2 as above and no files are transferred, whereas on other systems mbsync will successfully sync the files.

A secondary issue is that it is not helpful for the only error message to be printed to be of the form "... : Success." That might well be a problem with mbsync itself.

strace output attached (note: strace was run with 'Maildir-test' in place of 'Maildir' in the .mbsyncrc file since in the latter case the strace output exposes the names of all of my mail folders).

Microsoft Windows [Version 10.0.19041.450]

[mbsync_strace.txt](https://github.com/microsoft/WSL/files/5201999/mbsync_strace.txt)

**Comment by therealkenc** [comment, marked as answer]
Was able to sync my gmail (it's pulling down a decade of email in that screencap below). 

![image](https://user-images.githubusercontent.com/7049700/92848748-664ea080-f39f-11ea-82ef-63b08f182c86.png)

All WSL sees is a tcpv4 socket on port 993. There is no real path to a WSL actionable here. Moving to discussion, 
but you'll almost certainly get better eyeballs on this submission in a place where isync people hang out.

**Reply by AaronBWagner** [reply]
Thanks. mbsync works great for me on other systems --- it is only on WSL where I am having problems. That's why I posted here first. Granted, I don't have a standalone Ubuntu 20.04 system, so I can't do a head-to-head comparison. Perhaps this is an isync/ubuntu 20.04 issue that does not involve WSL. 

If I may ask: what happens if you run mbsync with the .mbsyncrc file I posted above? What Linux distribution and isync version are you using?

**Reply by therealkenc** [reply]
![image](https://user-images.githubusercontent.com/7049700/92961640-6d6bc200-f424-11ea-80da-426923d02fb2.png)

**Reply by AaronBWagner** [reply]
.mbsyncrc files are white-space sensitive and I messed up the white space in the original posting. It was supposed to be:

>  IMAPStore work-remote
> Host outlook.office365.com
> Port 993
> SSLType IMAPS
> SSLVersions TLSv1
> 
> MaildirStore work-local
> Path ~/Maildir/
> Inbox ~/Maildir/INBOX
> SubFolders Verbatim
> 
> Channel work
> Master :work-remote:
> Slave :work-local:
> Patterns "INBOX" 

That fixes all of the "unknown keywords." Sorry about that. 

I just installed Ubuntu 18.04 on WSL2 and I can confirm that mbsync works as expected there. So I assume this is an issue with Ubuntu 20.04 and/or mbsync. I will open a ticket elsewhere and mark this as answered. Thanks for the help.
